packages/blade/src/components/TextArea/_decisions/decisions.md

```diff
@@ -0,0 +1,70 @@
+| onChange             | `({ event, name, value }) => {}` |         | Function called when the value of the input field changes                                                                                                                                                                                                    |          |
+| onBlur               | `({ event, name, value }) => {}` |         | Function called when the input loses focus                                                                                                                                                                                                                   |          |
+| isDisabled           | `boolean`                        | `false` | Marks the input field as disabled. It'll also set aria-disabled to true and hence the input field will loose keyboard focus                                                                                                                                  |          |
+| isRequired           | `boolean`                        | `false` | Marks the input field as required. if the errorText is provided then that will be shown if the field is required                                                                                                                                             |          |
+| neccessityIndicator  | `required, optional`             |         | Indicator to be shown next to the label for emphasizing the necessity of the field. if optional is provided then it'll show `(optional)` and if required is provided then it'll show `*`                                                                     |          |
+| showClearButton      | `boolean`                        | `true`  | Defines if we want to show the clear icon button on the right side of the password field                                                                                                                                                                     |          |
+| onClearButtonClicked | `() => {}`                       |         | Used in conjunction with showClearButton prop. If clicked it removes the contents of the TextArea. Users can also hook their own analytics code or do something else.                                                                                        |          |
+| maxLines             | `number`                         | `2`     | Sets the number of lines (or rows) of the text area, minimum is `2` maximum is `5`                                                                                                                                                                           |          |
```

Comment:
for max line can we not set a upper limit ?

and can have a extra param call `autoGrow` which will automatically grow the text area to the `maxLines` value which consumer will specify

and if `autoGrow` is false we can honour the `maxLines` prop and stick the text area to that no of row only.

Reply:
Good idea! Yes we could add that but right now we've scoped this feature, until requirement/usecase arises. 

Check the Open Questions at the end.